src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DurableTask.AzureStorage;
+using DurableTask.AzureStorage.Tracking;
+using DurableTask.Core;
+using AzureStorage = DurableTask.AzureStorage;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// The Azure Storage implementation of additional methods not required by IOrchestrationService.
+    /// </summary>
+    internal class AzureStorageDurabilityProvider : DurabilityProvider
+    {
+        private readonly AzureStorageOrchestrationService serviceClient;
+
+        public AzureStorageDurabilityProvider(AzureStorageOrchestrationService service)
+            : base("Azure Storage", service, service)
+        {
+            this.serviceClient = service;
+        }
+
+        public override bool SupportsEntities => true;
+
+        /// <inheritdoc/>
+        public async override Task<IList<OrchestrationState>> GetAllOrchestrationStates(CancellationToken cancellationToken)
+        {
+            return await this.serviceClient.GetOrchestrationStateAsync(cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public async override Task<IList<OrchestrationState>> GetOrchestrationStateAsync(string instanceId, bool showHistory, bool showInput = true)
+        {
+            return await this.serviceClient.GetOrchestrationStateAsync(instanceId, allExecutions: false, fetchInput: showInput);
+        }
+
+        /// <inheritdoc/>
+        public async override Task RewindAsync(string instanceId, string reason)
+        {
+            await this.serviceClient.RewindTaskOrchestrationAsync(instanceId, reason);
+        }
+
+        /// <inheritdoc/>
+        public async override Task<IList<OrchestrationState>> GetAllOrchestrationStatesWithFilters(DateTime createdTimeFrom, DateTime? createdTimeTo, IEnumerable<OrchestrationRuntimeStatus> runtimeStatus, CancellationToken cancellationToken)
+        {
+            return await this.serviceClient.GetOrchestrationStateAsync(createdTimeFrom, createdTimeTo, runtimeStatus.Select(x => (OrchestrationStatus)x), cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public async override Task<string> RetrieveSerializedEntityState(string inputState)
+        {
+            // the input was compressed... read it back from blob
+            return await this.serviceClient.DownloadBlobAsync(inputState);
+        }
+
+        /// <inheritdoc/>
+        public async override Task<int> PurgeInstanceHistoryByInstanceId(string instanceId)
```

Comment:
Low-pri/optional: We may want to consider a richer return type. Otherwise it will require a new API if we want to return more info in the future.